src/modules/schnorr_adaptor/tests_impl.h

```diff
@@ -0,0 +1,991 @@
+    for (i = 0; i < COUNT; i++) {
+        adaptor_nonce_function_bip340_bitflip(args, 0, 32, algolen);
+        adaptor_nonce_function_bip340_bitflip(args, 1, 32, algolen);
+        adaptor_nonce_function_bip340_bitflip(args, 2, 32, algolen);
+        adaptor_nonce_function_bip340_bitflip(args, 3, 32, algolen);
+        /* Flip algo special case "SchnorrAdaptor/nonce" */
+        adaptor_nonce_function_bip340_bitflip(args, 4, algolen, algolen);
+        /* Flip algo again */
+        adaptor_nonce_function_bip340_bitflip(args, 4, algolen, algolen);
+        adaptor_nonce_function_bip340_bitflip(args, 5, 32, algolen);
+    }
+
+    /* NULL algo is disallowed */
+    CHECK(adaptor_nonce_function_bip340(nonce, msg, key, t, pk, NULL, 0, NULL) == 0);
+    CHECK(adaptor_nonce_function_bip340(nonce, msg, key, t, pk, algo, algolen, NULL) == 1);
+    /* Other algo is fine */
+    secp256k1_testrand_bytes_test(algo, algolen);
+    CHECK(adaptor_nonce_function_bip340(nonce, msg, key, t, pk, algo, algolen, NULL) == 1);
+
+    for (i = 0; i < COUNT; i++) {
+        unsigned char nonce2[32];
+        size_t algolen_tmp;
+
+        /* Different algolen gives different nonce */
+        uint32_t offset = secp256k1_testrand_int(algolen - 1);
+        algolen_tmp = (algolen + offset) % algolen;
+        CHECK(adaptor_nonce_function_bip340(nonce2, msg, key, t, pk, algo, algolen_tmp, NULL) == 1);
+        CHECK(secp256k1_memcmp_var(nonce, nonce2, 32) != 0);
+    }
+
+    /* NULL aux_rand argument is allowed, and identical to passing all zero aux_rand. */
+    memset(aux_rand, 0, 32);
+    CHECK(adaptor_nonce_function_bip340(nonce_z, msg, key, t, pk, algo, algolen, &aux_rand) == 1);
+    CHECK(adaptor_nonce_function_bip340(nonce, msg, key, t, pk, algo, algolen, NULL) == 1);
+    CHECK(secp256k1_memcmp_var(nonce_z, nonce, 32) == 0);
+}
+
+void test_schnorr_adaptor_api(void) {
+    unsigned char sk1[32];
+    unsigned char sk2[32];
+    unsigned char sk3[32];
+    unsigned char msg[32];
+    unsigned char t32[32];
+    unsigned char t[33] = {
+        0x02, 0xC6, 0x04, 0x7F, 0x94, 0x41, 0xED, 0x7D,
+        0x6D, 0x30, 0x45, 0x40, 0x6E, 0x95, 0xC0, 0x7C,
+        0xD8, 0x5C, 0x77, 0x8E, 0x4B, 0x8C, 0xEF, 0x3C,
+        0xA7, 0xAB, 0xAC, 0x09, 0xB9, 0x5C, 0x70, 0x9E, 0xE5
+    };
+    secp256k1_keypair keypairs[3];
+    secp256k1_keypair invalid_keypair = {{ 0 }};
+    secp256k1_xonly_pubkey pk[3];
+    secp256k1_xonly_pubkey zero_pk;
+    unsigned char sig[65];
+    unsigned char sig64[64];
+    unsigned char t2[33];
+    unsigned char adaptor[32];
+
+    /** setup **/
+    int ecount;
+
+    secp256k1_context_set_error_callback(CTX, counting_illegal_callback_fn, &ecount);
+    secp256k1_context_set_illegal_callback(CTX, counting_illegal_callback_fn, &ecount);
+    secp256k1_context_set_error_callback(STATIC_CTX, counting_illegal_callback_fn, &ecount);
+    secp256k1_context_set_illegal_callback(STATIC_CTX, counting_illegal_callback_fn, &ecount);
```

Comment:
This can all be removed. And instead of
```
CHECK(secp256k1_schnorr_adaptor_presign(CTX, sig, msg, &keypairs[0], t, NULL) == 1);
CHECK(ecount == 0);
```
write 
```
CHECK(secp256k1_schnorr_adaptor_presign(CTX, sig, msg, &keypairs[0], t, NULL) == 1);
```


Instead of 
```
CHECK(secp256k1_schnorr_adaptor_presign(STATIC_CTX, sig, msg, &keypairs[0], t, NULL) == 0);
CHECK(ecount == 1);
```
write
```
CHECK_ILLEGAL(STATIC_CTX, secp256k1_schnorr_adaptor_presign(STATIC_CTX, sig, msg, &keypairs[0], t, NULL) == 0);
```